src/control/cmd/daos/system.go

```diff
@@ -0,0 +1,76 @@
+//
+// (C) Copyright 2023 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package main
+
+/*
+#include "util.h"
+
+struct daos_rank_uri *
+get_rank_uri_at_idx(struct daos_rank_uri *uris, uint32_t i)
+{
+	return &uris[i];
+}
+*/
+import "C"
+
+import (
+	"github.com/pkg/errors"
+)
+
+type rankURI struct {
+	Rank int    `json:"rank"`
+	URI  string `json:"uri"`
+}
+
+type systemInfo struct {
+	Name     string     `json:"system_name"`
+	Provider string     `json:"fabric_provider"`
+	RankURIs []*rankURI `json:"rank_uris"`
+}
+
+type systemCmd struct {
+	Query systemQueryCmd `command:"query" description:"query DAOS system via the daos_agent"`
+}
+
+type systemQueryCmd struct {
+	daosCmd
+}
+
+func (cmd *systemQueryCmd) Execute(_ []string) error {
+	var cSysInfo *C.struct_daos_sys_info
+	rc := C.daos_mgmt_sys_info_alloc(&cSysInfo)
```

Comment:
Hmm. It seems a little strange to me to be doing a `get_attach_info()` in a function called `daos_mgmt_sys_info_alloc()`. Based on convention, I would expect something named `*_alloc()` to just be responsible for setting up a complex structure for use elsewhere. I wouldn't expect that helper function to be performing network operations. Maybe this ties into a wider conversation that I've been having with @liw over in [DAOS-13151](https://daosio.atlassian.net/browse/DAOS-13151)... I think that we may need some way to cache the `dc_mgmt_sys_info` data for the lifetime of the process.

The other thing to consider about this work in light of that other conversation is that the design of libdaos was originally intended to accommodate connections to multiple systems, as I understand it. Over time, though, assumptions about only connecting to a single system have been creeping in, so I wonder if we need to either fix those assumptions or else just formally abandon support for connections to multiple systems.